usr.sbin/devinfo/devinfo.c

```diff
@@ -181,16 +307,54 @@ int
 print_rman_resource(struct devinfo_res *res, void *arg __unused)
 {
 	struct devinfo_dev	*dev;
-	
-	printf("    ");
-	print_resource(res);
+	struct devinfo_rman	*rman;
+	rman_res_t		end;
+	char			*res_str, *entry;
+	bool			hexmode;
+
 	dev = devinfo_handle_to_device(res->dr_device);
+	rman = devinfo_handle_to_rman(res->dr_rman);
+	hexmode =  (rman->dm_size > 1000) || (rman->dm_size == 0);
+	end = res->dr_start + res->dr_size - 1;
+
+	if (hexmode) {
+		if (res->dr_size > 1)
+			asprintf(&res_str, "0x%jx-0x%jx", res->dr_start, end);
+		else
+			asprintf(&res_str, "0x%jx", res->dr_start);
+	}
+	else {
+		if (res->dr_size > 1)
+			asprintf(&res_str, "%ju-%ju", res->dr_start, end);
+		else
+			asprintf(&res_str, "%ju", res->dr_start);
+	}
+
+	xo_emit("{P:    }");
+
+	if (last_res == NULL) {
+		// First resource
+		xo_open_list(res_str);
+	}
+	else if (strcmp(res_str, last_res) != 0) {
```

Comment:
'} else if' is how this should be formatted, on one line.

Reply:
Fixed